Source/Dafny/ConstructorWarning.cs

```diff
@@ -44,6 +44,15 @@ class ConstructorWarningVisitor : TopDownVisitor<Unit> {
     }
     return base.VisitOneExpr(expr, ref st);
   }
+  protected override bool VisitOneStmt(Statement stmt, ref Unit st) {
+    if (stmt is NestedMatchStmt matchStmt) {
+      var matchStmtCases = matchStmt.Cases;
```

Comment:
Since this variable is used just once, you can inline it: either by changing the `foreach` below to refer to `matchStmt.cases` or by changing the pattern match above to bind `Cases` (look up the documentation on pattern matching if you're not sure how to do that)